addons/website/static/src/js/content/inject_dom.js

```diff
@@ -15,6 +15,25 @@ export function unhideConditionalElements() {
     document.head.appendChild(styleEl);
     const conditionalEls = document.querySelectorAll('[data-visibility="conditional"]');
     for (const conditionalEl of conditionalEls) {
+        // In the case of a mega menu block, hide the mega menu link in the bar according to the same rule
+        // Change the visibility for Desktop and Mobile views
+        if (conditionalEl.classList.contains("s_mega_menu_odoo_menu")) {
+            const megaMenuDesktop = conditionalEl.closest("li[role~='presentation']");
+            const menusDesktop = megaMenuDesktop.parentElement.querySelectorAll('li');
+            // Get the index of the mega menu in the menu list
+            let index = -1;
```

Comment:
I think basing the implementation on the index is not robust enough: what if the user has some custo which adds some menu items in mobile or inverting them?

I would check how we do for moving the mega menu content in mobile. We are probably matching the desktop and mobile locations another way... hopefully.

Reply:
In the "menu.js" script, from L787 to L806 there is the "_moveMegaMenu" function that does the matching using the index of the element.